apps/ex_nvr/lib/ex_nvr/authorization.ex

```diff
@@ -0,0 +1,11 @@
+defmodule ExNVR.Authorization do
+  @moduledoc """
+  Basic user authorization module by role.
+  """
+  alias ExNVR.Accounts.User
+  def authorized?(%User{role: :admin}, :device, _action), do: {:ok, :authorized}
+  def authorized?(%User{role: :user}, :device, :read), do: {:ok, :authorized}
+  def authorized?(_user, :recording, :read), do: {:ok, :authorized}
+
+  def authorized?(_user, _resource, _action), do: {:error, :unauthorized}
```

Comment:
Just make this functions follow Elixir spec
```elixir
def authorize(%User{role: :admin}, _resource, _action), do: :ok
def authorize(%User{role: :user}, _resource, :read), do: :ok
def authorize(_user, _resource, _action), do: {:error, :unauthorized}
```
I guess this is the only rules you need for now, once we add more resources, we'll update this functions.